pkg/env/env_test.go

```diff
@@ -0,0 +1,53 @@
+package env
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/iver-wharf/wharf-core/internal/testutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func testBind(t *testing.T, ptr interface{}, envKey string, envValue string, want interface{}) {
+	t.Run(envKey, func(t *testing.T) {
+		testutil.Setenv(t, envKey, envValue)
+		require.NoError(t, Bind(ptr, envKey))
+		// dereference the pointer inside the interface,
+		// turning e.g. interface{*int} into interface{int}
+		got := reflect.ValueOf(ptr).Elem().Interface()
+		assert.Equal(t, want, got)
+	})
+}
+
+func TestBind(t *testing.T) {
+	var (
+		myString   string
+		myBool     bool
+		myInt      int
+		myInt32    int32
+		myInt64    int64
+		myUint     uint
+		myUint32   uint32
+		myUint64   uint64
+		myFloat32  float32
+		myFloat64  float64
+		myDuration time.Duration
+	)
+	testBind(t, &myString, "MY_STR", "bar", "bar")
+	testBind(t, &myBool, "MY_BOOL", "true", true)
+	testBind(t, &myInt, "MY_INT", "-123", int(-123))
+	testBind(t, &myInt32, "MY_INT32", "-123", int32(-123))
+	testBind(t, &myInt64, "MY_INT64", "-123", int64(-123))
+	testBind(t, &myUint, "MY_UINT", "123", uint(123))
+	testBind(t, &myUint32, "MY_UINT32", "123", uint32(123))
+	testBind(t, &myUint64, "MY_UINT64", "123", uint64(123))
+	testBind(t, &myFloat32, "MY_FLOAT32", "123.0", float32(123.0))
+	testBind(t, &myFloat64, "MY_FLOAT64", "123.0", float64(123.0))
+	testBind(t, &myDuration, "MY_DURATION", "5s", 5*time.Second)
+}
+
+func TestBindMultiple_noErrorOnNilMap(t *testing.T) {
```

Comment:
Here is another case of vaiolation of caps standardisation. 
```suggestion
func TestBindMultipleNoErrorOnNilMap(t *testing.T) {
```

Reply:
Sorry, force of habit from the naming rules for example tests, where they have

```go
func Example{NameOfTypeOrFunction}_{comment}
func Example{NameOfType}_{NameOfMemberForSaidType}_{comment}
```

<https://blog.golang.org/examples#TOC_4.>

Where the compiler actually gives a warning when you do it wrong.

Open question: Should we use the same naming style for tests?

Reply:
Ok cool i had no idea that was a thing anywhere. I guess we keep it then. It make my eyes hurt a bit but if its on the golan blog and well documented it seems like the purpose is justified.

Reply:
We have no defined convention for unit test names, and I've been writing a lot of example tests lately. The only rule for unit tests is that they start with `Test` and that the next character is uppercase.

I believe it was Ewelina who suggested something before that they used in some of their C# projects of

```go
func Test{NameOfTypeOrMember}_{StateOfObject}_{ExpectedResult}(t *testing.T)

// For example
func TestBindMultiple_GivenNil_DoesNothing(t *testing.T)
```

but that idea didn't get adopted as it was quite verbose.

It's not a topic we've dug into really for Wharf. I've created an issue on it so we could tackle it later if we want to: https://github.com/iver-wharf/iver-wharf.github.io/issues/64

OK if I resolve this comment without applying your change?

Reply:
Resolving as you did mark your review as "Approved"